### PR TITLE
Clean Up Predicate A Bit

### PR DESCRIPTION
**Pull request by stephen-lazaro**
Alias existing functions from symbolic ones
Add contravariant instance
Guarantee all instances serializable

For #202.

### CODE REVIEW

File: core/src/main/scala/cats/collections/Predicate.scala [hunk 1]
```diff
@@ -63,12 +63,20 @@ object Predicate extends PredicateInstances {
...
+    override def unit: Predicate[Unit] = Predicate.empty
+  }
+
   implicit def predicateMonoid[A]: Monoid[Predicate[A]] = new Monoid[Predicate[A]] {
     override def empty: Predicate[A] = Predicate.empty
     override def combine(l: Predicate[A], r: Predicate[A]): Predicate[A] = l union r
   }
 
-  implicit val predicateInstance: MonoidK[Predicate] = new MonoidK[Predicate] {
+  implicit val predicateMonoidK: MonoidK[Predicate] = new MonoidK[Predicate] {
```
Comment (stephen-lazaro) [inline core/src/main/scala/cats/collections/Predicate.scala]: Dunno if this rename is cool from a bincompat perspective.

Comment (larsrh) [reply core/src/main/scala/cats/collections/Predicate.scala]: There are no guarantees about that yet, so it's fine.

Comment (johnynek) [reply core/src/main/scala/cats/collections/Predicate.scala]: I think we should add mima checking sooner than later.

I personally really avoid libraries that don't try to minimize breaks (and in fact, despite contributing to this library, I am afraid to depend on it).

File: core/src/main/scala/cats/collections/Predicate.scala [hunk 2]
```diff
@@ -63,12 +63,20 @@ object Predicate extends PredicateInstances {
 }
 
 trait PredicateInstances {
+  implicit def predicateContravariantMonoidal: ContravariantMonoidal[Predicate] = new ContravariantMonoidal[Predicate] {
+    override def contramap[A, B](fb: Predicate[A])(f: B => A): Predicate[B] =
+      Predicate(f andThen fb.apply)
+    override def product[A, B](fa: Predicate[A], fb: Predicate[B]): Predicate[(A, B)] =
+      Predicate(v => fa(v._1) || fb(v._2))
```
Comment (larsrh) [inline core/src/main/scala/cats/collections/Predicate.scala]: I'm curious about the choice of `||` here, what's the reasoning?

File: core/src/main/scala/cats/collections/Predicate.scala (hunk 1 above)
Comment (larsrh) [reply core/src/main/scala/cats/collections/Predicate.scala]: Happy to call it 1.0 at some point in the near future.

File: core/src/main/scala/cats/collections/Predicate.scala (hunk 2 above)
Comment (stephen-lazaro) [reply core/src/main/scala/cats/collections/Predicate.scala]: Yeah, it's definitely a bit arbitrary, but I made the choice based on having the monoidal product act similarly to the monoids already defined for this type. `||` gives us union, which is probably the more in line with user expectations given the monoids are also union.

Comment (larsrh) [reply core/src/main/scala/cats/collections/Predicate.scala]: Could you please add law tests for `ContravariantMonoidal`? Then I think we're good to go.

Comment (stephen-lazaro) [reply core/src/main/scala/cats/collections/Predicate.scala]: Sure, I'll get to that today / tomorrow probably.

Comment (stephen-lazaro) [reply core/src/main/scala/cats/collections/Predicate.scala]: I did it with kind of stupid `Eq` instances, but I'm not sure there are better ones. If there's a standard way to do `Eq` for functions, let me know and I can change it up.

**Comment by larsrh** [comment]
Thanks @stephen-lazaro!

**Pull request merged**